### PR TITLE
Updated Atlassian package repository address

### PR DESCRIPTION
**Pull request by pmckeown**
The Atlassian package repository has been moved.  While this address still works with a 301 permanent redirection, anyone using this file as documentation for allow-listing egress to these endpoints (like me) needs the accurate addresses.  Since this one has been moved by Atlassian, I thought I'd send this PR to update it at source.
 

```shell
curl -v  "https://maven.atlassian.com/content/repositories/atlassian-public/"
*   Trying 104.192.143.13:443...
* Connected to maven.atlassian.com (104.192.143.13) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-ECDSA-AES128-GCM-SHA256
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=AU; ST=New South Wales; L=Sydney; O=Atlassian Pty Ltd; CN=*.atlassian.com
*  start date: Apr  8 00:00:00 2020 GMT
*  expire date: Jun 10 12:00:00 2022 GMT
*  subjectAltName: host "maven.atlassian.com" matched cert's "*.atlassian.com"
*  issuer: C=US; O=DigiCert Inc; OU=www.digicert.com; CN=DigiCert SHA2 High Assurance Server CA
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x7fa6bb012200)
> GET /content/repositories/atlassian-public/ HTTP/2
> Host: maven.atlassian.com
> user-agent: curl/7.77.0
> accept: */*
> 
< HTTP/2 301 
< date: Tue, 26 Apr 2022 06:24:00 GMT
< content-type: text/html
< content-length: 169
< server: globaledge-envoy
< location: https://packages.atlassian.com/content/repositories/atlassian-public/
< x-envoy-upstream-service-time: 811
< expect-ct: report-uri="https://web-security-reports.services.atlassian.com/expect-ct-report/artifactory", max-age=86400
< strict-transport-security: max-age=63072000; preload
< x-content-type-options: nosniff
< x-xss-protection: 1; mode=block
< atl-traceid: 377b6acbbd651d6b
< report-to: {"group": "endpoint-1", "max_age": 600, "endpoints": [{"url": "https://dj9s4kmieytgz.cloudfront.net"}], "include_subdomains": true}
< nel: {"report_to": "endpoint-1", "max_age": 600, "include_subdomains": true, "failure_fraction": 0.001}
< 
<html>
<head><title>301 Moved Permanently</title></head>
<body>
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx/1.18.0</center>
</body>
</html>
* Connection #0 to host maven.atlassian.com left intact
```

### CODE REVIEW

**Comment by AbdelHajou** [comment]
Should we consider adding this to the update script as well? Changing this file only applies to documentation but doesn't affect the actual URL used

**Comment by pmckeown** [comment]
> Should we consider adding this to the update script as well? Changing this file only applies to documentation but doesn't affect the actual URL used

Yeah - good call.  Will update that too.

File: src/main/java/org/dependencytrack/persistence/DefaultObjectGenerator.java [hunk 1]
```diff
@@ -238,7 +238,7 @@ private void loadDefaultRepositories() {
             qm.createRepository(RepositoryType.GEM, "rubygems.org", "https://rubygems.org/", true, false);
             qm.createRepository(RepositoryType.HEX, "hex.pm", "https://hex.pm/", true, false);
             qm.createRepository(RepositoryType.MAVEN, "central", "https://repo1.maven.org/maven2/", true, false);
-            qm.createRepository(RepositoryType.MAVEN, "atlassian-public", "https://maven.atlassian.com/content/repositories/atlassian-public/", true, false);
+            qm.createRepository(RepositoryType.MAVEN, "atlassian-public", "https://packages.atlassian.com/content/repositories/atlassian-public/", true, false);
```
Comment (AbdelHajou) [inline src/main/java/org/dependencytrack/persistence/DefaultObjectGenerator.java:241]: Would this apply to existing installations? The createRepository function checks whether a repository for this type and identifier already exists. Maybe this should be added to the v450Updater as well.

Comment (nscuro) [reply src/main/java/org/dependencytrack/persistence/DefaultObjectGenerator.java:241]: You're correct @AbdelHajou. This change should be made in `v450Updater` as well.

@pmckeown See here for how it was done in v3.8.0 for a change in the Maven Central URL: https://github.com/DependencyTrack/dependency-track/blob/3.8.0/src/main/java/org/dependencytrack/upgrade/v380/v380Updater.java#L41

Comment (pmckeown) [reply src/main/java/org/dependencytrack/persistence/DefaultObjectGenerator.java:241]: Great point. Hadn’t considered that. Will take at the update for 3.8.0 tomorrow.

**Comment by nscuro** [comment]
Thanks @pmckeown! The failing test is due to a change I recently made.

**Pull request merged**